stablehlo/dialect/TypeInference.cpp

```diff
@@ -1246,8 +1246,22 @@ LogicalResult inferReduceWindowOp(
   return success();
 }
 
-LogicalResult inferReturnOp(Optional<Location> location,
-                            SmallVectorImpl<Type>& inferredReturnTypes) {
+LogicalResult inferReturnOp(Optional<Location>, SmallVectorImpl<Type>&) {
+  return success();
+}
+
+LogicalResult inferScatterOp(Optional<Location>, ValueRange inputs,
+                             SmallVectorImpl<Type>& inferredReturnTypes) {
+  for (auto input : inputs) {
+    Type elementTy = getElementTypeOrSelf(input.getType());
+    auto rankedTy = input.getType().dyn_cast<RankedTensorType>();
+    if (!rankedTy) {
+      inferredReturnTypes.emplace_back(UnrankedTensorType::get(elementTy));
+      continue;
+    }
+
+    inferredReturnTypes.emplace_back(input.getType());
+  }
```

Comment:
Also this loop can be replaced by `llvm::append_range(inferredReturnTypes, inputs.getTypes());`.

Reply:
Nice! Fixed.